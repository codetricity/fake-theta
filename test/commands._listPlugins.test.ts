import { request, toMatchInvalidHeaderParameterError } from './helper';

describe('POST /osc/commands/execute _listPlugins', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._listPlugins' })
      .expect(200);
    expect(res.body.results.plugins).toMatchSnapshot();
  });
});

describe('THETA X POST /osc/commands/execute _listPlugins', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._listPlugins' })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body.results.plugins).toMatchSnapshot();
  });
});

describe('THETA Z1 POST /osc/commands/execute _listPlugins', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._listPlugins' })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body.results.plugins).toMatchSnapshot();
  });
});

describe('Unknown Type POST /osc/commands/execute _listPlugins', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera._listPlugins' })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
