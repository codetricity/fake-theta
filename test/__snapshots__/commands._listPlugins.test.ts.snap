// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute _listPlugins should respond with results 1`] = `
[
  {
    "boot": true,
    "exitStatus": "success",
    "force": true,
    "foreground": true,
    "message": "",
    "packageName": "com.example.mytestapplication1",
    "pluginName": "MyTestApplication1",
    "running": false,
    "type": "user",
    "version": "1.1",
    "webServer": false,
  },
  {
    "boot": true,
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.example.mytestapplication2",
    "pluginName": "MyTestApplication2",
    "running": false,
    "type": "user",
    "version": "1.2",
    "webServer": false,
  },
  {
    "boot": true,
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.example.mytestapplication3",
    "pluginName": "MyTestApplication3",
    "running": false,
    "type": "user",
    "version": "1.3",
    "webServer": false,
  },
  {
    "boot": true,
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.example.mytestapplication4",
    "pluginName": "MyTestApplication4",
    "running": false,
    "type": "user",
    "version": "1.4",
    "webServer": false,
  },
  {
    "boot": true,
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.example.mytestapplication5",
    "pluginName": "MyTestApplication5",
    "running": false,
    "type": "user",
    "version": "1.5",
    "webServer": false,
  },
]
`;

exports[`THETA X POST /osc/commands/execute _listPlugins should respond with results 1`] = `
[
  {
    "boot": true,
    "exitStatus": "success",
    "force": true,
    "foreground": true,
    "message": "",
    "packageName": "com.example.mytestapplication1",
    "pluginName": "MyTestApplication1",
    "running": false,
    "type": "user",
    "version": "1.1",
    "webServer": false,
  },
  {
    "boot": true,
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.example.mytestapplication2",
    "pluginName": "MyTestApplication2",
    "running": false,
    "type": "user",
    "version": "1.2",
    "webServer": false,
  },
  {
    "boot": true,
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.example.mytestapplication3",
    "pluginName": "MyTestApplication3",
    "running": false,
    "type": "user",
    "version": "1.3",
    "webServer": false,
  },
  {
    "boot": true,
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.example.mytestapplication4",
    "pluginName": "MyTestApplication4",
    "running": false,
    "type": "user",
    "version": "1.4",
    "webServer": false,
  },
  {
    "boot": true,
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.example.mytestapplication5",
    "pluginName": "MyTestApplication5",
    "running": false,
    "type": "user",
    "version": "1.5",
    "webServer": false,
  },
]
`;

exports[`THETA Z1 POST /osc/commands/execute _listPlugins should respond with results 1`] = `
[
  {
    "boot": true,
    "bootOptions": "",
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.theta.remoteplayback",
    "pluginName": "Remote Playback",
    "running": false,
    "type": "system",
    "version": "2.10.3",
    "webServer": false,
  },
  {
    "boot": true,
    "bootOptions": "",
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.theta360.remotecontrol",
    "pluginName": "Remote Control",
    "running": false,
    "type": "system",
    "version": "1.0.0",
    "webServer": false,
  },
  {
    "boot": true,
    "bootOptions": "",
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.theta360.usbstorage",
    "pluginName": "USB data transfer",
    "running": false,
    "type": "system",
    "version": "3.0.0",
    "webServer": true,
  },
  {
    "boot": false,
    "bootOptions": "",
    "exitStatus": "success",
    "force": false,
    "foreground": false,
    "message": "",
    "packageName": "com.example.sample",
    "pluginName": "SampleApplication",
    "running": true,
    "type": "user",
    "version": "0.0.1",
    "webServer": false,
  },
]
`;
